lab/src/Microsoft.ServiceModel.Syndication/src/Atom10FeedFormatter.cs

```diff
@@ -243,162 +223,281 @@ internal static void ReadCategory(XmlReader reader, SyndicationCategory category
         }
 
-        internal bool TryParseFeedElementFrom(XmlReader reader, SyndicationFeed result)
+        internal async Task<bool> TryParseFeedElementFromAsync(XmlReaderWrapper reader, SyndicationFeed result)
         {
-            if (reader.IsStartElement(Atom10Constants.AuthorTag, Atom10Constants.Atom10Namespace))
-            {
-                result.Authors.Add(ReadPersonFrom(reader, result));
-            }
-            else if (reader.IsStartElement(Atom10Constants.CategoryTag, Atom10Constants.Atom10Namespace))
-            {
-                result.Categories.Add(ReadCategoryFrom(reader, result));
-            }
-            else if (reader.IsStartElement(Atom10Constants.ContributorTag, Atom10Constants.Atom10Namespace))
-            {
-                result.Contributors.Add(ReadPersonFrom(reader, result));
-            }
-            else if (reader.IsStartElement(Atom10Constants.GeneratorTag, Atom10Constants.Atom10Namespace))
-            {
-                result.Generator = reader.ReadElementString();
-            }
-            else if (reader.IsStartElement(Atom10Constants.IdTag, Atom10Constants.Atom10Namespace))
-            {
-                result.Id = reader.ReadElementString();
-            }
-            else if (reader.IsStartElement(Atom10Constants.LinkTag, Atom10Constants.Atom10Namespace))
-            {
-                result.Links.Add(ReadLinkFrom(reader, result));
-            }
-            else if (reader.IsStartElement(Atom10Constants.LogoTag, Atom10Constants.Atom10Namespace))
-            {
-                result.ImageUrl = new Uri(reader.ReadElementString(), UriKind.RelativeOrAbsolute);
-            }
-            else if (reader.IsStartElement(Atom10Constants.RightsTag, Atom10Constants.Atom10Namespace))
-            {
-                result.Copyright = ReadTextContentFrom(reader, "//atom:feed/atom:rights[@type]");
-            }
-            else if (reader.IsStartElement(Atom10Constants.SubtitleTag, Atom10Constants.Atom10Namespace))
-            {
-                result.Description = ReadTextContentFrom(reader, "//atom:feed/atom:subtitle[@type]");
-            }
-            else if (reader.IsStartElement(Atom10Constants.TitleTag, Atom10Constants.Atom10Namespace))
-            {
-                result.Title = ReadTextContentFrom(reader, "//atom:feed/atom:title[@type]");
-            }
-            else if (reader.IsStartElement(Atom10Constants.UpdatedTag, Atom10Constants.Atom10Namespace))
-            {
-                reader.ReadStartElement();
-                result.LastUpdatedTime = DateFromString(reader.ReadString(), reader);
-                reader.ReadEndElement();
-            }
-            else
+
+            string name = reader.LocalName;
+            string ns = reader.NamespaceURI;
+
+            if(ns == Atom10Constants.Atom10Namespace)
             {
-                return false;
-            }
-            return true;
+                switch (name)
+                {
```

Comment:
This code isn't equivalent to what you replaced.  IsStartElement calls MoveToContent, checks that the node type is XmlNodeType.Element. You need to do the following:
```csharp
if(await reader.MoveToContentAsync() != XmlNodeType.Element)
{
  return false;
}

string name = reader.LocalName;
string ns = reader.NamespaceURI;
```